src/rpc/mining.cpp

```diff
@@ -132,7 +132,9 @@ static UniValue generateBlocks(const CScript& coinbase_script, int nGenerate, ui
             continue;
         }
         std::shared_ptr<const CBlock> shared_pblock = std::make_shared<const CBlock>(*pblock);
-        if (!ProcessNewBlock(Params(), shared_pblock, true, nullptr))
+        CValidationState state;
+        ProcessNewBlock(Params(), shared_pblock, state, true).wait();
+        if (ChainActive().Tip()->GetBlockHash() != pblock->GetHash())
```

Comment:
A wild block could be mined in between?

Reply:
Right, but most likely something was mined before the new block got accepted, which is the right error for that. That said, I think there's a bug here that ChainActive().Tip() needs cs_main, which isn't held here.